.github/scripts/parse_rpt.py

```diff
@@ -102,11 +105,48 @@
 
 ## Result File Check
 if _generator_is['sky130hvl_ldo']:
-   json_filename = "spec.json"
+    json_filename = "spec.json"
 else:
-   json_filename = "test.json"
+    json_filename = "test.json"
 
 if check_gen_files(json_filename, _generator_is, cryo_library):
-        print("Flow check is clean!")
+    print("Flow check is clean!")
 else:
     print("Flow check failed!")
+
+if len(sys.argv) == 1:
+    sim_state_filename = "work/sim_state_file.txt"
+
+    if check_ngspice_version():
+        result_filename = "work/prePEX_sim_result" 
+
+        template_filename = "../../../.github/scripts/expected_sim_outputs/prePEX_sim_result.txt"
+        with open(result_filename) as f2, open(template_filename) as f1:
+            content1 = f2.readlines()
+            content2 = f1.readlines()
+            if content1 != content2:
+                warnings.warn("Simulation result file does not match! Please contact a maintainer of the repo!", DeprecationWarning)
+    else:
+        print("The ngspice version does not match, "
```

Comment:
I think we should do the frequency and ngspice checks separately. A different ngspice version might still produce the correct results.

Reply:
Yeah that's a nice idea, Thanks!